$NetBSD: patch-toolkit_library_moz.build,v 1.3 2015/02/28 04:30:55 ryoon Exp $

MOZ_GSTREAMER:
  GSTREAMER_LIBS are linked to libxul on Darwin, while they are
  dlopen(3)'ed at runtime on other platforms. The problem is that the
  toolkit being cocoa isn't relevant at all. It's Darwin that needs
  the special handling, not Cocoa.

--- toolkit/library/moz.build.orig	2015-02-17 21:40:48.000000000 +0000
+++ toolkit/library/moz.build
@@ -214,6 +214,7 @@ if CONFIG['OS_ARCH'] == 'Linux' and CONF
     ]
 
 OS_LIBS += CONFIG['MOZ_CAIRO_OSLIBS']
+OS_LIBS += CONFIG['MOZ_LIBV4L2_LIBS']
 OS_LIBS += CONFIG['MOZ_WEBRTC_X11_LIBS']
 
 if CONFIG['MOZ_NATIVE_JPEG']:
@@ -225,6 +226,27 @@ if CONFIG['MOZ_NATIVE_PNG']:
 if CONFIG['MOZ_NATIVE_HUNSPELL']:
     OS_LIBS += CONFIG['MOZ_HUNSPELL_LIBS']
 
+if CONFIG['MOZ_NATIVE_OGG']:
+    OS_LIBS += CONFIG['MOZ_OGG_LIBS']
+
+if CONFIG['MOZ_NATIVE_THEORA']:
+    OS_LIBS += CONFIG['MOZ_THEORA_LIBS']
+
+if CONFIG['MOZ_NATIVE_VORBIS']:
+    OS_LIBS += CONFIG['MOZ_VORBIS_LIBS']
+
+if CONFIG['MOZ_NATIVE_TREMOR']:
+    OS_LIBS += CONFIG['MOZ_TREMOR_LIBS']
+
+if CONFIG['MOZ_NATIVE_OPUS']:
+    OS_LIBS += CONFIG['MOZ_OPUS_LIBS']
+
+if CONFIG['MOZ_NATIVE_SPEEX']:
+    OS_LIBS += CONFIG['MOZ_SPEEX_LIBS']
+
+if CONFIG['MOZ_NATIVE_SOUNDTOUCH']:
+    OS_LIBS += CONFIG['MOZ_SOUNDTOUCH_LIBS']
+
 if CONFIG['MOZ_NATIVE_LIBEVENT']:
     OS_LIBS += CONFIG['MOZ_LIBEVENT_LIBS']
 
@@ -234,8 +256,14 @@ if CONFIG['MOZ_NATIVE_LIBVPX']:
 if not CONFIG['MOZ_TREE_PIXMAN']:
     OS_LIBS += CONFIG['MOZ_PIXMAN_LIBS']
 
-if CONFIG['MOZ_ALSA']:
-    OS_LIBS += CONFIG['MOZ_ALSA_LIBS']
+if CONFIG['MOZ_NATIVE_GRAPHITE2']:
+    OS_LIBS += CONFIG['MOZ_GRAPHITE2_LIBS']
+
+if CONFIG['MOZ_NATIVE_HARFBUZZ']:
+    OS_LIBS += CONFIG['MOZ_HARFBUZZ_LIBS']
+
+if CONFIG['MOZ_OSS']:
+    OS_LIBS += CONFIG['MOZ_OSS_LIBS']
 
 if CONFIG['HAVE_CLOCK_MONOTONIC']:
     OS_LIBS += CONFIG['REALTIME_LIBS']
@@ -364,7 +392,7 @@ if CONFIG['OS_ARCH'] == 'WINNT':
 if CONFIG['MOZ_ENABLE_QT']:
     OS_LIBS += CONFIG['XEXT_LIBS']
 
-if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'cocoa' and CONFIG['MOZ_GSTREAMER']:
+if CONFIG['OS_ARCH'] == 'Darwin' and CONFIG['MOZ_GSTREAMER']:
     OS_LIBS += CONFIG['GSTREAMER_LIBS']
 
 if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'windows':
